scheduler/graphql_mid/scalars.py

```diff
@@ -37,6 +38,11 @@ def parse_specific_site(site: str):
         # Single site case
         return frozenset([parse_specific_site(sites)])
 
+def parse_origins(name: str) -> Origin:
+    try:
+        return Origins[name].value
```

Comment:
Then here we could do:
```python
try:
    return Origins[name].value()
```
to invoke the lambda and created the desired one, and all should work as expected, I think.